vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,34 +10,177 @@
+      winrt::Microsoft::ReactNative::implementation::ReactNativeHost::GetReactNativeHost(m_context.Properties());
+  if (!host) {
+    return;
+  }
+
+  RegisterWndClass(); // creates and register a windows class
+
+  if (!m_hwnd) {
+    auto CompositionHwndHost = winrt::Microsoft::ReactNative::CompositionHwndHost();
+    winrt::Microsoft::ReactNative::ReactViewOptions viewOptions;
+    viewOptions.ComponentName(L"Modal");
+    CompositionHwndHost.ReactViewHost(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::MakeViewHost(host, viewOptions));
+    HINSTANCE hInstance = GetModuleHandle(NULL);
+    winrt::impl::abi<winrt::Microsoft::ReactNative::ICompositionHwndHost>::type *pHost{nullptr};
+    winrt::com_ptr<::IUnknown> spunk;
+    CompositionHwndHost.as(spunk);
+
+    // get the root hwnd
+    auto roothwnd = reinterpret_cast<HWND>(
+        winrt::Microsoft::ReactNative::ReactCoreInjection::GetTopLevelWindowId(m_context.Properties().Handle()));
+
+    m_hwnd = CreateWindow(
+        c_modalWindowClassName,
+        L"React-Native Modal",
+        WS_OVERLAPPEDWINDOW,
+        CW_USEDEFAULT,
+        CW_USEDEFAULT,
+        MODAL_DEFAULT_WIDTH,
+        MODAL_DEFAULT_HEIGHT,
+        roothwnd, // parent
+        nullptr,
+        hInstance,
+        spunk.get());
+
+    // Check if window creation succeeded
+    if (!m_hwnd) {
+      throw std::exception("Failed to create new hwnd for Modal: " + GetLastError());
+    }
+
+    spunk.detach();
+  }
+
+  ShowWindow(m_hwnd, SW_NORMAL);
+  BringWindowToTop(m_hwnd);
+  SetFocus(m_hwnd);
+}
+
+void WindowsModalHostComponentView::HideOnUIThread() noexcept {
+  if (m_hwnd) {
+    ::ShowWindow(m_hwnd, SW_HIDE);
+  }
+}
+
+// Windows Procedure - callback function used for handling all messages (generated by NTUser or manual calls to
+// SendMessage)
+LRESULT CALLBACK ModalBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+  auto data = reinterpret_cast<::IUnknown *>(GetProp(
+      hwnd,
+      CompHostProperty)); // gets data handle from the property list of specified window (ie the window we want to make)
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result) {
+      return result;
+    }
+  }
+
+  switch (message) {
+    case WM_NCCREATE: { // sent before WM_CREATE, lparam should be identical to members of CreateWindowEx
+      auto createStruct = reinterpret_cast<CREATESTRUCT *>(lparam); // CreateStruct
+      auto windowData = static_cast<::IUnknown *>(createStruct->lpCreateParams);
+      SetProp(hwnd, CompHostProperty, reinterpret_cast<::IUnknown *>(windowData)); // adds new properties to window
```

Comment:
Re-reading this function... isn't the intent that `windowData` is the same field as `data`? So... does this make sense instead?
```suggestion
      data = static_cast<::IUnknown *>(createStruct->lpCreateParams);
      SetProp(hwnd, CompHostProperty, data); // adds new properties to window
```